bash_completion

```diff
@@ -2614,6 +2612,17 @@ _complete_as_root()
     [[ $EUID -eq 0 || ${root_command-} ]]
 }
 
+# Complete on available commands, subject to `no_empty_cmd_completion`.
+#
+# @since 2.12
+_comp_compgen_commands()
+{
+    [[ ! ${cur-} ]] && shopt -q no_empty_cmd_completion && return
+    _comp_compgen -- -c -o plusdirs
+    # for e.g. spaces in paths to and in command names
+    ((${#COMPREPLY[@]} == 0)) || compopt -o filenames
```

Comment:
```suggestion
    # -o filenames for e.g. spaces in paths to and in command names
    _comp_compgen -- -c -o plusdirs && compopt -o filenames
```

For the CI error.

Reply:
I now realized that I haven't added `@return` in the doc comment of `_comp_compgen`, but `_comp_compgen -- [options]` is intended to succeed only when at least one completion is generated, and this behavior is already used in the codebase.

It is mentioned in [api-and-naming.md in #993](https://github.com/scop/bash-completion/pull/993/files#diff-0cb3af15071817f620202dd47344251e6a8bff22af238511159fb03ef2e886f0R161-R163). I added https://github.com/scop/bash-completion/pull/993/commits/ed8e81e246d20c8ecf7f943dfe7f8e41060504ac in #993.